dlrover/python/elastic_agent/torch/training.py

```diff
@@ -452,6 +453,52 @@ def __init__(
             training_log_file, failure_node_errors
         )
 
+    @prof
+    def _stop_workers_ascend(self, worker_group: WorkerGroup) -> None:
+        logger.info("stop workers via SIGKILL for Ascend NPU")
+        """The ASCEND framework might fork multiple sub-processes, we should
+        stop all the children processes before shutdown the workers
+        """
+        """print out a snapshot of all processes"""
+        env_utils.print_process_list()
+
+        if self._pcontext is not None:
+            pc_pids = set(self._pcontext.pids().values())
+            logger.info(f"try to kill child processes of {pc_pids}")
+            for pid in pc_pids:
+                logger.info(f"kill child processes of process {pid}")
+                try:
+                    pp = psutil.Process(pid)
+                    cp = pp.children()
+                    for proc in cp:
+                        logger.info(f"kill sub {proc.pid} of parent {pid}")
+                        os.kill(proc.pid, signal.SIGKILL)
+                except Exception as e:
+                    logger.info(f"kill {pid}: {str(e)}")
+
+        self._shutdown(death_sig=signal.SIGKILL)
+
+        """cleanup orphan processes if exists"""
+        self._stop_orphan_workers(worker_group)
+
+        """print out a snapshot of all processes again"""
+        env_utils.print_process_list()
+
+    @prof
+    def _stop_orphan_workers(self, wg: WorkerGroup) -> None:
+        """How we define the orphan workers
+        1. ppid == 1
+        2. is_worker_process() is True
+        """
+        try:
+            for p in psutil.process_iter():
+                if p.ppid() == 1 and env_utils.is_worker_process(p.pid):
+                    name = " ".join(p.cmdline())
+                    logger.info(f"find orphan workers {p.pid}: {name}")
+                    os.kill(p.pid, signal.SIGKILL)
```

Comment:
In Ascend NPU, we prefer to send SIGKILL directly since SIGTERM often doesn't work. It is also compatible with previous code in ascend.npu cases

![image](https://github.com/user-attachments/assets/e9191b41-6122-4c30-abb9-90ba659292db)